tests/test_version_range.py

```diff
@@ -150,3 +152,19 @@ def test_GemVersionRange_from_native_range_with_pessimistic_operator(self):
             VersionConstraint(comparator=">=", version=RubygemsVersion(string="2.0.8")),
             VersionConstraint(comparator="<", version=RubygemsVersion(string="2.1")),
         )
+
+
+@pytest.mark.parametrize("scheme, range_class", RANGE_CLASS_BY_SCHEMES.items())
+class TestVersionRangeImplementations:
+    VERSION_RANGES_BY_SCHEMES = {
+        "nginx": ["0.8.40+", "0.7.52-0.8.39", "0.9.10", "1.5.0+, 1.4.1+"],
+    }
+
+    def test_from_native_and_from_string_equality(self, scheme, range_class):
+        if scheme not in self.VERSION_RANGES_BY_SCHEMES:
```

Comment:
As mentioned above, you should not iterate and skip if not present. Instead iterate on the actual tests  and not the range class... and have a test that fails if we do not cover all the schemes.

Reply:
But this test will almost certainly fail as we have many unimplemented VersionRange

Reply:
not if marked as expected failure

Reply:
Thanks!